core/src/write.jl

```diff
@@ -376,11 +376,14 @@ function subgrid_level_table(
     (; t, saveval) = saved.subgrid_level
     subgrid = integrator.p.subgrid
 
-    nelem = length(subgrid.subgrid_id)
+    nelem = length(subgrid.level)
     ntsteps = length(t)
 
     time = repeat(datetime_since.(t, config.starttime); inner = nelem)
-    subgrid_id = repeat(subgrid.subgrid_id; outer = ntsteps)
+    subgrid_id = repeat(
+        sort(vcat(subgrid.subgrid_id_static, subgrid.subgrid_id_time));
+        outer = ntsteps,
```

Comment:
Time comes from the SavedValues that is used by the SavingCallback, with one t for each save. There are no duplicates in there, hence the repeat. I added a test line to ensure that for each subgrid ID each timestamp is unique, because we don't want to save the same time twice.